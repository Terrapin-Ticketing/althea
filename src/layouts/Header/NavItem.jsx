import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router'
import styled from 'styled-components'

const NavItemContainer = styled.li`
  cursor: pointer;
  float: left;
  padding: 20px 30px;
  font-size: 16px;
  font-weight: 700;
`

const StyledLink = styled(Link)`
  &:focus, &:active {
    outline: 0;
  }
`;

const NavItem = (props) => (
  <NavItemContainer>
    <StyledLink onClick={props.onClick ? () => props.onClick() : null} to={props.href}>{props.label}</StyledLink>
  </NavItemContainer>
)

NavItem.propTypes = {
  href: PropTypes.string,
  label: PropTypes.string,
  onClick: PropTypes.func
}

export default NavItem
