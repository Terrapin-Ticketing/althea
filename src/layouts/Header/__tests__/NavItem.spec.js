import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import NavItem from '../NavItem'

describe('NavItem Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NavItem href='help' label='Help' />)
    expect(wrapper).toMatchSnapshot()
  })
})
