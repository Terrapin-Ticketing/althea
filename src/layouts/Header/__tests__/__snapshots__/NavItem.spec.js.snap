// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavItem
    href="help"
    label="Help"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="help"
      >
        Help
      </Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Help",
        "onlyActiveOnIndex": false,
        "style": Object {},
        "to": "help",
      },
      "ref": null,
      "rendered": "Help",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="help"
        >
          Help
        </Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Help",
          "onlyActiveOnIndex": false,
          "style": Object {},
          "to": "help",
        },
        "ref": null,
        "rendered": "Help",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
